providers/base/bin/switch_power_mode.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+# This file is part of Checkbox.
+#
+# Copyright 2024 Canonical Ltd.
+#    Authors: Bin Li <bin.li@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+# pylint: disable=consider-using-f-string
+
+""" Switching the power mode to check if the power mode can be switched. """
+from pathlib import Path
+import subprocess
+import contextlib
+
+
+def get_sysfs_content(path):
+    """
+    Reads the content of a sysfs file.
+    Args:
+        path (pathlib.Path): Path to the sysfs file.
+    Raises:
+        SystemExit: If the file could not be read.
+    """
+    with open(path, "rt", encoding="utf-8") as stream:
+        content = stream.read().strip()
+    if not content:
+        raise SystemExit("Failed to read sysfs file: {}".format(path))
+    return content
+
+
+def set_power_profile(profile):
+    """
+    Sets the power profile to the specified value.
+    Args:
+        profile (str): The power profile to set (e.g., "power-saver").
+    Raises:
+        SystemExit: If the power profile could not be set.
+    """
+    profile = "power-saver" if profile == "low-power" else profile
```

Comment:
Where does this come from?

Reply:
The value in /sys and powerprofile is different, so I need convert it before set mode with powerprofilectl.

```
$ cat /sys/firmware/acpi/platform_profile_choices 
low-power balanced performance

$ powerprofilesctl list 
  performance:
  balanced:
* power-saver:
 ```

Reply:
I added a comment above this code.
```
diff --git a/providers/base/bin/switch_power_mode.py b/providers/base/bin/switch_power_mode.py
index 81811fa31..e17073885 100755
--- a/providers/base/bin/switch_power_mode.py
+++ b/providers/base/bin/switch_power_mode.py
@@ -48,6 +48,8 @@ def set_power_profile(profile):
     Raises:
         SystemExit: If the power profile could not be set.
     """
+    # In sys file the modes are low-power, balanced, or performance
+    # but powerprofilesctl only accepts power-saver, balanced or performance
     profile = "power-saver" if profile == "low-power" else profile
     try:
         subprocess.check_call(["powerprofilesctl", "set", profile])
```